data_encryption/models/encrypted_data.py

```diff
@@ -0,0 +1,130 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+import json
+
+from odoo import api, fields, models, SUPERUSER_ID
+from odoo.exceptions import AccessError, ValidationError
+from odoo.tools.config import config
+from odoo.tools.translate import _
+from odoo.tools import ormcache
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from cryptography.fernet import Fernet, InvalidToken
+except ImportError as err:  # pragma: no cover
+    _logger.debug(err)
+
+
+class EncryptedData(models.Model):
+    """Model to store encrypted data by environment (prod, preprod...)"""
+
+    _name = 'encrypted.data'
+    _description = 'Store any encrypted data by environment'
+
+    name = fields.Char(
+        required=True, readonly=True, index=True,
+        help="Technical name")
+    environment = fields.Char(
+        required=True,
+        index=True,
+        help="Concerned Odoo environment (prod, preprod...)")
+    encrypted_data = fields.Binary()
+
+    _sql_constraints = [
+        ('name_environment_uniq', 'unique (name, environment)',
+         'You can not store multiple encrypted data for the same record and \
+          environment')
+    ]
+
+    @api.model
+    @ormcache('self._uid', 'name', 'env')
+    def _get(self, name, env=None):
```

Comment:
can we give such methods better meaningful names? Eg: `_encrypted_get` `_encrypted_get_json` `_encrypted_store` etc?

Reply:
I don't really see the point as you are inside the EncryptedData class, so it seems clear that the _get will read/decrypt the data ans the _store will write/encrypt data... no?

Reply:
@florian-dacosta because it makes easier to understand what it does, especially if you inspect model's methods via python autocomplete/debugger.

Reply:
@simahawk Well I am not really convinced about this one, even if you inspect, you know on which model you are and 1_get/_store seems clean enough, adding _encrypted seems redondant. But anyway, I just did it by courtesy, I don't really mind, so let's do this as there is not yet module dependent on it yet!